src/jshint.js

```diff
@@ -2218,6 +2217,19 @@ var JSHINT = (function() {
     return this;
   }));
   prefix("new", function() {
+    var mp = metaProperty("target");
+    if (mp) {
+      if (mp.value === "target") {
+        if (!state.inESNext(true)) {
+          warning("W119", state.tokens.prev, "new.target");
+        }
+        if (!state.inClassBody) {
```

Comment:
if metaProperty only returns if value === "target", JSHint would throw a ton of errors when it encounters invalid meta properties because it would treat them as construction.